### PR TITLE
UIOR-183 Allow 0.0 for unit price in Cost for Orders

### PR DESCRIPTION
**Pull request by aliaksei-chumakou**
## Purpose
Allow 0.0 for unit price in Cost for Orders.
https://issues.folio.org/browse/UIOR-183

## Approach
Basically, there is a different validations on UI for quantities and prices, since required prices can be 0 or `0.00` and required quantities should be more or equal 1

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/POLine/Cost/CostForm.js [hunk 1]
```diff
@@ -24,16 +23,33 @@ import {
...
 
+// Validation for Fields with type 'number' requires positive integer
+export const requiredPositiveQuantity = (value) => {
+  return value >= 1
+    ? undefined
+    : <FormattedMessage id="ui-orders.cost.validation.shouldBePositive" />;
+};
+
+const validateRequiredNotNegative = (value) => {
+  return value === 0 || value > 0
```
Comment (craigmcnally) [inline src/components/POLine/Cost/CostForm.js]: could this be shortened to simply ` value < 0 ? <formattedMessage ...> : undefined`?  Was it done this way for consistency with the other ternary expressions used in the other validation functions?

Comment (aliaksei-chumakou) [reply src/components/POLine/Cost/CostForm.js]: basically in JS the way you've suggested is doing a quite different comparison. we are not sure if value is a `Number` type. So empty string will pass through as a good value in your case

Comment (craigmcnally) [reply src/components/POLine/Cost/CostForm.js]: Thanks makes sense now... but doesn't that mean that `'0'` would be considered invalid, while `0` or `'2'` are considered valid?

Comment (aliaksei-chumakou) [reply src/components/POLine/Cost/CostForm.js]: That's true, but in overall `return value === 0 || value > 0` precisely describes what we expect as truthy values, so for readability purposes I prefer to leave it as is. 
If you look into `parse={parseMoney}` field attribute, which calls before validate, it return `Number` or some falsy value (empty string included), what we filter here as not valid. Hope this makes sense!

Comment (NikitaSedyx) [reply src/components/POLine/Cost/CostForm.js]: hm, what about `value >= 0`?

Comment (NikitaSedyx) [reply src/components/POLine/Cost/CostForm.js]: doesn't work with empty string, but fine for undefined

**Comment by NikitaSedyx** [comment]
:+1:

File: src/components/POLine/Cost/CostForm.js (hunk 1 above)
Comment (craigmcnally) [reply src/components/POLine/Cost/CostForm.js]: Right, `value === 0 || value > 0` gets us 99% of the way there, the case I mention is just an edge case.  I'm fine with leaving this as-is for now.  Thanks for helping me understand guys

**Pull request merged**